components/next_steps_view/steps.ts

```diff
@@ -6,45 +6,93 @@ import {getLicense} from 'mattermost-redux/selectors/entities/general';
 import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
 import {UserProfile} from 'mattermost-redux/types/users';
 
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {isEqual} from 'lodash';
+
 import {GlobalState} from 'types/store';
 import {RecommendedNextSteps, Preferences} from 'utils/constants';
 import {localizeMessage} from 'utils/utils';
 
 import CompleteProfileStep from './steps/complete_profile_step';
+import SetupPreferencesStep from './steps/setup_preferences_step/setup_preferences_step';
 import InviteMembersStep from './steps/invite_members_step';
 import TeamProfileStep from './steps/team_profile_step';
+import EnableNotificationsStep from './steps/enable_notifications_step/enable_notifications_step';
 
 export type StepComponentProps = {
     id: string;
     currentUser: UserProfile;
     onSkip: (id: string) => void;
     onFinish: (id: string) => void;
 }
-
 export type StepType = {
     id: string;
     title: string;
     component: React.ComponentType<StepComponentProps>;
-}
+
+    // An array of all roles a user must have in order to see the step e.g. admins are both system_admin and system_user
+    // so you would require ['system_admin','system_user'] to match.
+    // to show step for all roles, leave the roles array blank.
+    roles: Array<string>;
+};
 
 export const Steps: StepType[] = [
     {
         id: RecommendedNextSteps.COMPLETE_PROFILE,
-        title: localizeMessage('next_steps_view.titles.completeProfile', 'Complete your profile'),
+        title: localizeMessage(
+            'next_steps_view.titles.completeProfile',
+            'Complete your profile'
+        ),
         component: CompleteProfileStep,
+        roles: [],
     },
     {
         id: RecommendedNextSteps.TEAM_SETUP,
-        title: localizeMessage('next_steps_view.titles.teamSetup', 'Name your team'),
+        title: localizeMessage(
+            'next_steps_view.titles.teamSetup',
+            'Name your team'
+        ),
+        roles: ['system_admin', 'system_user'],
```

Comment:
Shouldn't this one and the one below it only be `system_admin`?

Reply:
The way the role system returns your roles is you have your role, and all roles underneath you. So an admin has roles: `system_admin system_user`, while a user has just `system_user`. There's a comment here: https://github.com/mattermost/mattermost-webapp/blob/0ce08414374cc97dc109f64e4717e7b970107da5/components/next_steps_view/steps.ts#L33 about it. 

I was trying to find a clean solution to this for a while, and this way meant the least conditional logic, otherwise it got weird when there were certain cases of a step that needed to be for BOTH admins and normal users (ie, the complete profile step). A simple isAdmin boolean wouldn't work for those cases.

Reply:
Gotcha, I wasn't aware how the roles worked. Thanks for the insight :)